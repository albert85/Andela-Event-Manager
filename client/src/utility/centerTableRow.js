import React, { Component } from 'react';

class CenterRows extends Component {
  render() {
    return (

                <tr className="border border-white">
                    <td scope="row">{ this.props.i + 1 }</td>
                    <td>{this.props.centers.name}</td>
                    <td>{this.props.centers.location}</td>
                    <td>{this.props.centers.amount}</td>
                    <td>{this.props.centers.capacity}</td>
                    <td>
                        <div className="row">
                            <div className="col mb-2">
                                <button type="button" onClick = { this.props.handleCenterDetails.bind(this, this.props.centerId) } className="btn btn-primary btn-block">
                                <i className="fa fa-pencil" aria-hidden="true"> Edit</i>
                                </button>
                            </div>
                                <div className="col">
                                    <button type="button" className="btn btn-success btn-block" onClick = { this.props.handleEditCenterDetails.bind(this, this.props.centerId) }>
                                        <i className="fa fa-book" aria-hidden="true"> Update</i>
                                    </button>
                                </div>

                            </div>

                    </td>

                </tr>

    );
  }
}

export default CenterRows;
