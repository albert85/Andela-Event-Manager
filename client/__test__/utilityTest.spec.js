import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import centerTableRow from '../src/utility/centerTableRow';

let isComponentMounted;

configure({ adapter: new Adapter() });


describe('Index Routes', () => {
  const props = {
    centers: {
      name: '',
      location: '',
      amount: '',
      capacity: '',
    },
  };

  const getComponentStatus = () => {
    if (!isComponentMounted) {
      isComponentMounted = shallow(<centerTableRow {...props} />);
    }
    return isComponentMounted;
  };

  it('should successfully rendered', () => {
    expect(getComponentStatus()).toMatchSnapshot();
  });

  it('should successfully rendered', () => {
    console.log(getComponentStatus());
    expect(getComponentStatus()).toHaveLength(1);
  });
});
