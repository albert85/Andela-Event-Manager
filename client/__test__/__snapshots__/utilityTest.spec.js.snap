// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Routes should successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <centerTableRow
    centers={
      Object {
        "amount": "",
        "capacity": "",
        "location": "",
        "name": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "centers": Object {
        "amount": "",
        "capacity": "",
        "location": "",
        "name": "",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "centerTableRow",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "centers": Object {
          "amount": "",
          "capacity": "",
          "location": "",
          "name": "",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "centerTableRow",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
